### PR TITLE
[SPARK-11197] [SQL] run SQL on files directly

### PR DESCRIPTION
**Pull request by davies**
This PR introduce a new feature to run SQL directly on files without create a table, for example:

```
select id from json.`path/to/json/files` as j
```

### CODE REVIEW

**Comment by rxin** [comment]
The syntax looks good to me. Couple comments:

1. Let's add a config variable so users can disable this.
2. Make sure the error messages make sense. e.g. if there is a database named "json", the error message should say "there is no table or file named xxx"

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 1]
```diff
@@ -49,6 +49,9 @@ trait CheckAnalysis {
     plan.foreachUp {
       case p if p.analyzed => // Skip already analyzed sub-plans
 
+      case u: UnresolvedRelation =>
+        u.failAnalysis(s"Table Not Found: ${u.tableIdentifier}")
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala]: Table not found

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 2]
```diff
@@ -1782,6 +1782,34 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
...
+
+    val e2 = intercept[AnalysisException] {
+      sql("select * from no_db.no_table")
+    }
+    assert(e2.message.contains("Table Not Found"))
+
+    val e3 = intercept[AnalysisException] {
+      sql("select * from json.invalid_file")
+    }
+    assert(e3.message.contains("No input paths specified"))
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: This error message seems confusing - there was a path specified?

Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: This is the existing behavior, different data source will have different messages. This error message came from Hadoop, I think.

Comment (marmbrus) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: `No valid files found at specified path: '$path'`?

Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: I filed https://issues.apache.org/jira/browse/SPARK-11220

**Comment by marmbrus** [comment]
Thanks!  Merging to master.

**Pull request closed without merging**